c/test/test_for.cpp

```diff
@@ -0,0 +1,204 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda_runtime.h>
+
+#include <algorithm>
+
+#include "c2h.h"
+#include <cccl/c/for.h>
+
+void for_each(cccl_iterator_t input, unsigned long long num_items, cccl_op_t op)
+{
+  cudaDeviceProp deviceProp;
+  cudaGetDeviceProperties(&deviceProp, 0);
+
+  const int cc_major = deviceProp.major;
+  const int cc_minor = deviceProp.minor;
+
+  const char* cub_path        = TEST_CUB_PATH;
+  const char* thrust_path     = TEST_THRUST_PATH;
+  const char* libcudacxx_path = TEST_LIBCUDACXX_PATH;
+  const char* ctk_path        = TEST_CTK_PATH;
+
+  cccl_device_for_build_result_t build;
+  REQUIRE(
+    CUDA_SUCCESS
+    == cccl_device_for_build(&build, input, op, cc_major, cc_minor, cub_path, thrust_path, libcudacxx_path, ctk_path));
+  const std::string sass = inspect_sass(build.cubin, build.cubin_size);
+  // REQUIRE(sass.find("LDL") == std::string::npos);
+  // REQUIRE(sass.find("STL") == std::string::npos);
```

Comment:
suggestion: let's return these checks:

```suggestion
  REQUIRE(sass.find("LDL") == std::string::npos);
  REQUIRE(sass.find("STL") == std::string::npos);
```